include/xsimd/types/xsimd_utils.hpp

```diff
@@ -240,6 +240,7 @@ namespace xsimd
 
             template <typename... Ts>
             using index_sequence_for = make_index_sequence<sizeof...(Ts)>;
+        #endif
```

Comment:
Symbols below should be available no matter c++ version.
Original code only defines those symbols under c++11, and causes error when build xsimd neon if -std=c++14.
`xsimd::detail::int_sequence’ has not been declared`

Reply:
The `int_sequence` code looks super redundant with the `index_sequence` code. I'll propose a patch that make the whole stuff more generic and take your remark into account.